integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/SqlDWDatabricksVisitor.java

```diff
@@ -0,0 +1,114 @@
+ * ((select \"id\" FROM dbo.exampleInputA WHERE postalCode != '55555') q)</code>) then the name will
+ * be COMPLEX.
+ */
+public class SqlDWDatabricksVisitor<D extends OpenLineage.Dataset>
+    extends QueryPlanVisitor<LogicalPlan, D> {
+  private final DatasetFactory<D> factory;
+  private static final Pattern dbJdbcPattern = Pattern.compile("database=([^;]*);?");
+  private static final Pattern serverJdbcPattern = Pattern.compile("jdbc:([^;]*);?");
+  private static final String DATABRICKS_CLASS_NAME = "com.databricks.spark.sqldw.SqlDWRelation";
+
+  public SqlDWDatabricksVisitor(OpenLineageContext context, DatasetFactory<D> factory) {
+    super(context);
+    this.factory = factory;
+  }
+
+  public static boolean hasSqlDWDatabricksClasses() {
+    try {
+      SqlDWDatabricksVisitor.class.getClassLoader().loadClass(DATABRICKS_CLASS_NAME);
+      return true;
+    } catch (Exception e) {
+      // swallow- we don't care
+    }
+    return false;
+  }
+
+  protected boolean isSqlDwRelationClass(LogicalPlan plan) {
+    return plan instanceof LogicalRelation
+        && ((LogicalRelation) plan).relation().getClass().getName().equals(DATABRICKS_CLASS_NAME);
+  }
+
+  @Override
+  public boolean isDefinedAt(LogicalPlan plan) {
+    return isSqlDwRelationClass(plan);
+  }
+
+  private String getName(BaseRelation relation) {
+    String tableName;
+    try {
+      tableName = (String) FieldUtils.readField(relation, "tableNameOrSubquery", true);
+    } catch (IllegalAccessException e) {
+      throw new RuntimeException("Unable to discover SQLDW tableNameOrSubquery property");
+    }
+    // The Synapse connector will return a table name wrapped in double quotes
+    // or you could have a query string (e.g. (SELECT * FROM table)q)
+    if (tableName.startsWith("\"") && tableName.endsWith("\"")) {
+      tableName = tableName.replace("\"", "");
+    }
+    // TODO If there is a query, we should ultimately parse the SQL but
+    // returning COMPLEX to be consistent with other implementations.
+    if (tableName.startsWith("(")) {
+      tableName = "COMPLEX";
+    }
+    return tableName;
+  }
+
+  private String getNameSpace(BaseRelation relation) {
+
+    String jdbcUrl;
+    try {
+      Object fieldDetails = FieldUtils.readField(relation, "params", true);
+      jdbcUrl = (String) MethodUtils.invokeMethod(fieldDetails, true, "jdbcUrl");
+    } catch (IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+      throw new RuntimeException("Unable to discover SQLDW jdbcUrl Parameters");
+    }
+
+    Matcher serverMatcher = serverJdbcPattern.matcher(jdbcUrl);
+    boolean serverIsFound = serverMatcher.find();
+    Matcher dbMatcher = dbJdbcPattern.matcher(jdbcUrl);
+    boolean dbIsFound = dbMatcher.find();
+
+    if (!(dbIsFound && serverIsFound)) {
+      throw new RuntimeException(
+          "Unable to discover SQLDW database name or server name from jdbc url");
+    }
+
```

Comment:
Same comment as above - log a warning and let's avoid the exception

Reply:
Thank you again for reviewing! Resolved in latest update.